go/vt/vtgate/engine/insert.go

```diff
@@ -243,11 +244,57 @@ func (ins *Insert) TryExecute(vcursor VCursor, bindVars map[string]*querypb.Bind
 
 // TryStreamExecute performs a streaming exec.
 func (ins *Insert) TryStreamExecute(vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool, callback func(*sqltypes.Result) error) error {
-	res, err := ins.TryExecute(vcursor, bindVars, wantfields)
+	if ins.Opcode != InsertSelect {
+		res, err := ins.TryExecute(vcursor, bindVars, wantfields)
```

Comment:
Shouldn't primitive to primitive calls always go through the vcursor?

Reply:
earlier was not needed as the call came through vcursor itself and was delegated, here may be the case.